engine/userguide/eng-image/multistage-build.md

```diff
@@ -0,0 +1,145 @@
+---
+description: Keeping your images small with multi-stage images
+keywords: images, containers, best practices
+title: Use multi-stage builds
+---
+
+{% include edge_only.md section="page" %}
+
+Multi-stage builds are a new feature in Docker 17.05, and they will be exciting
+to anyone who has struggled to optimize Dockerfiles while keeping them easy to
+read and maintain.
+
+> **Acknowledgment**:
+> Special thanks to Alex Ellis for granting permission to use his blog post
+> [Builder pattern vs. Multi-stage builds in Docker](http://blog.alexellis.io/mutli-stage-docker-builds/)
+> as the basis of the examples below.
+
+## Before multi-stage builds
+
+One of the most challenging things about building images is keeping the image
+size down. Each instruction in the Dockerfile adds a layer to the image, and you
+need to remember to clean up any artifacts you don't need before moving on to
+the next layer. To write a really efficient Dockerfile, you have traditionally
+needed to employ shell tricks and other logic to keep the layers as small as
+possible and to ensure that each layer has the artifacts it needs from the
+previous layer and nothing else.
+
+It was actually very common to have one Dockerfile to use for development (which
+contained everything needed to build your application), and a slimmed-down one
+to use for production, which only contained your application and exactly what
+was needed to run it. This has been referred to as the "builder
+pattern".Maintaining two Dockerfiles is not ideal.
```

Comment:
Should there be a space here between `pattern".M` ?

+pattern".Maintaining two Dockerfiles is not ideal.